app/src/ui/lib/text-box.tsx

```diff
@@ -271,6 +274,14 @@ export class TextBox extends React.Component<ITextBoxProps, ITextBoxState> {
           this.inputElement.blur()
         }
       }
+    } else if (
+      this.props.type === 'search' &&
+      event.key === 'Enter' &&
+      value !== undefined &&
+      value !== '' &&
+      this.props.onEnterClicked !== undefined
```

Comment:
Would just checking `this.props.onEnterClicked` would be sufficient without the `!== undefined`?

Reply:
Thank you for jumping in and reviewing this! Our team has a code convention of not using falsy comparisons. Technically, it would work without the `!== undefined` but we prefer to be explicit. The only time we do that is if it is a boolean that type-wise cannot be undefined.

Reply:
Good deal, thank you @tidy-dev!